### PR TITLE
[BB-3540] [BD-27] [TNL-7890] Investigate PDF issues in Common Cartridge Course Conversions

### PR DESCRIPTION
**Pull request by shimulch**
This PR copies files that are outside the ``web_resources`` folder in ``imscc`` into ``static`` folder in ``olx``.

**JIRA tickets**: 
- https://tasks.opencraft.com/browse/BB-3540

**Discussions**: None

**Dependencies**: None

**Screenshots**: N/A

**Sandbox URL**: N/A

**Merge deadline**: None

**Testing instructions**:

1. Clone this PR
2. Find a common cartridge file that has PDF or other assets outside ``web_resources`` folder.
3. Convert the CC file
4. Check if the file has been copied to the ``static`` directory in the output compressed file.

**Author notes and concerns**:
N/A

**Reviewers**
- [ ] @farhaanbukhsh 
- [ ] edX reviewer[s] TBD

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @shimulch! I've created [BLENDED-735](https://openedx.atlassian.net/browse/BLENDED-735) to keep track of it in Jira.   More details are on the [BD-27](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1544912903/BD-27+Common+Cartridge+Converter) project page.


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by shimulch** [comment]
It is asking for reformatting files that are not related to this PR. So I will run the ``black`` reformat after review. I hope that is ok.

**Comment by alangsto** [comment]
@shimulch I just merged a PR that should solve some of the formatting issues, if you rebase from master it should pass now

**Comment by shimulch** [comment]
@alangsto Thanks. Formatting issues are gone after rebase.

File: src/cc2olx/models.py [hunk 1]
```diff
@@ -282,6 +286,16 @@ def get_resource_content(self, identifier):
                     )
                 )
                 return "html", {"html": html}
+            elif "web_resources" not in str(res_filename):
+                # Keep track of this file and later copy into static dir
```
Comment (farhaanbukhsh) [inline src/cc2olx/models.py:290]: can we make this comment a little more elaborate?

Comment (farhaanbukhsh) [reply src/cc2olx/models.py:290]: Like what exactly it means if the  web_resource folder is not present.

File: src/cc2olx/models.py [hunk 2]
```diff
@@ -282,6 +286,16 @@ def get_resource_content(self, identifier):
...
+                self.extra_static_files.append(res_relative_path)
+                html = (
+                    '<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>'
+                    '</head><body><p><a href="{}" alt="{}">{}<a></p></body></html>'.format(
+                        "/static/" + res_relative_path, res_relative_path, res_relative_path
```
Comment (farhaanbukhsh) [inline src/cc2olx/models.py:295]: Nit: `/static/` can be moved in a constant in this class and then use everywhere, else there can be a chance in future if we change the path we have to do multiple places

File: src/cc2olx/models.py (hunk 1 above)
Comment (farhaanbukhsh) [inline src/cc2olx/models.py:290]: we can elaborate on this comment as to why it needs to be done in this way.

**Review by farhaanbukhsh — CHANGES_REQUESTED** [review]
There are few nits, else this looks good. Can you also increase the version number?

File: src/cc2olx/models.py (hunk 1 above)
Comment (shimulch) [reply src/cc2olx/models.py]: Comment updated with more clear explanation.

File: src/cc2olx/models.py (hunk 2 above)
Comment (shimulch) [reply src/cc2olx/models.py]: Moved ``static`` to ``OLX_STATIC_DIR`` const.

File: src/cc2olx/models.py (hunk 1 above)
Comment (shimulch) [reply src/cc2olx/models.py]: Comment updated.

File: src/cc2olx/main.py [hunk 3]
```diff
@@ -27,13 +27,18 @@ def convert_one_file(input_file, workspace, link_file=None):
...
-        ],
-    )
+    file_list = [
+        (str(olx_filename), "course.xml"),
+        (str(cartridge.directory / "web_resources"), "/static/"),
+    ]
+
+    # Add static files that are outside of web_resources directory
+    file_list += [
+        (str(cartridge.directory / filepath), "/static/{}".format(filepath))
```
Comment (farhaanbukhsh) [inline src/cc2olx/main.py:37]: Nit: We can make use of  `OLX_STATIC_DIR` here just to keep it consistent throughout. what do you say?

**Review by farhaanbukhsh — APPROVED** [review]
👍 

I have approved this with just one Nit, else it looks good to me.

- [ ] I tested this: Create a imscc file with resource placed outside the web_resource directory, ran the the tool on the file, imported it in the course. The link seems to be functional.
- [x] I read through the code
~~- [ ] I checked for accessibility issues~~
- [x] Includes documentation
~~- [ ] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~~

File: src/cc2olx/main.py (hunk 3 above)
Comment (shimulch) [reply src/cc2olx/main.py:37]: Yes, I agree, thanks :-)

**Review by alangsto — APPROVED** [review]

**Pull request merged**